transcom/mymove
# MB-12328 Edit PPM Shipment Office

## [Jira ticket](https://dp3.atlassian.net/browse/MB-12328) for this change

## Summary
This PR updates the ghc shipment update handler to use the new orchestrator service object. I also refactored the `MTOShipmentUpdater` service object to use validators rather than creating separate update functions for customer/office/prime users and removed the `RetrieveMTOShipment`.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps



1. Access the office app 
2. Log in as a service counselor
3. Look at the `PPMSC1` move and get the moveTaskOrderID, shipment id, and eTag from the console
4. Using postman, set the `If-Match` header and PATCH http://officelocal:3000/ghc/v1/move_task_orders/:moveTaskOrderID/mto_shipments/:shipmentID with this body 
```
{
  "ppmShipment": {
    "expectedDepartureDate": "2022-06-28",
    "actualMoveDate": "2022-05-11",
    "pickupPostalCode": "90210",
    "secondaryPickupPostalCode": "90002",
    "destinationPostalCode": "10108",
    "secondaryDestinationPostalCode": "79329",
    "sitExpected": true,
    "sitLocation": "ORIGIN",
    "sitEstimatedWeight": 222,
    "sitEstimatedEntryDate": "2022-06-02",
    "sitEstimatedDepartureDate": "2022-07-07",
    "estimatedWeight": 1111,
    "netWeight": 3333,
    "hasProGear": true,
    "proGearWeight": 555,
    "spouseProGearWeight": 33
    }       
}
```
5. Make sure the update works as expected and note the eTag
6. Because of the way estimated incentive is calculated, advance updates should be done separately.
7. Make a second PATCH to http://officelocal:3000/ghc/v1/move_task_orders/:moveTaskOrderID/mto_shipments/:shipmentID with this body 
```
{
  "ppmShipment": {
    "advance": 12345,
    "advanceRequested": true
    }       
}
```

## Verification Steps for Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.




### Backend
